From 230415e97590feeeae37579a6f429bfe1aa983ae Mon Sep 17 00:00:00 2001
From: Anthony Loiseau <anthony.loiseau@allcircuits.com>
Date: Tue, 19 Oct 2021 11:42:59 +0200
Subject: [PATCH] PNG generation: fix ignored compression level

User-wanted compression level is printed in verbose traces
but libpng was always given Z_BEST_COMPRESSION (9) instead
of this user-wanted compression level.
---
 fbgrab.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/fbgrab.c b/fbgrab.c
index 42731c6..576938d 100644
--- a/fbgrab.c
+++ b/fbgrab.c
@@ -351,7 +351,7 @@ static void write_PNG(unsigned char *outbuffer, char *filename,
 
     png_init_io(png_ptr, outfile);
 
-    png_set_compression_level(png_ptr, Z_BEST_COMPRESSION);
+    png_set_compression_level(png_ptr, compression);
 
     bit_depth = 8;
     color_type = PNG_COLOR_TYPE_RGB_ALPHA;
